src/tests/ftest/network/zero_config.py

```diff
@@ -0,0 +1,152 @@
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+import os
+from avocado import fail_on
+from apricot import TestWithServers
+from daos_racer_utils import DaosRacerCommand
+from command_utils import CommandFailure
+from general_utils import pcmd, get_host_data
+
+
+class ZeroConfigTest(TestWithServers):
+    """Test class for zero-config tests.
+
+    Test Class Description:
+        Test to verify that client application to libdaos can access a running
+        DAOS system with & without any special environment variable definitions.
+
+    :avocado: recursive
+    """
+
+    def get_port_cnt(self, hosts, dev, port_counter):
+        """Get the port count info for device names specified.
+
+        Args:
+            hosts (list): list of hosts
+            dev (str): device to get counter information for.
+            port_counter (str): port counter to get information from
+
+        Returns:
+            dict: a dictionary of data values for each NodeSet key
+
+        """
+        b_path = "/sys/class/infiniband"
+        file = os.path.join(b_path, "ports/1/counters", port_counter)
+        # Check if if exists for the host
+        check_cmd = "test -f {}".format(file)
+        result = pcmd(hosts, check_cmd, True, None, 0)
+        if len(result) > 1 or 0 not in result:
+            self.fail("Could not find {} on {}".format(
+                file, self.hostlist_clients[0]))
+
+        cmd = "cat {}".format(file)
+        text = "port_counter"
+        error = "Error obtaining {} info".format(port_counter)
+        return get_host_data(hosts, cmd, text, error, 20)
+
+    @fail_on(CommandFailure)
+    def verify_client_run(self, server_idx, exp_iface, env=False):
+        """Verify the interface assigned by runnning a libdaos client.
+
+        Args:
+            server_idx (int): server to get config from.
+            exp_iface (str): expected iterface to check.
+            env (bool): add OFI_INTERFACE variable to exported variables of
+                client command. Defaults to False.
+
+        Returns:
+            bool: returns status
+
+        """
+        hfi_map = {"ib0": "hfi1_0", "ib1": "hfi1_1"}
+
+        # Get counter values for hfi devices before and after
+        cnt_before = self.get_port_cnt(
+            self.hostlist_clients[0], hfi_map[exp_iface], "port_rcv_data")
+
+        # Let's run daos_racer as a client
+        daos_racer = DaosRacerCommand(self.bin, self.hostlist_clients[0])
+        daos_racer.get_params(self)
+
+        # Update env_name list to add OFI_INTERFACE if needed.
+        if env:
+            self.update_env_names(["OFI_INTERFACE"])
```

Comment:
(pylint-no-member) Instance of 'ZeroConfigTest' has no 'update_env_names' member